app/Http/Controllers/V2/Dashboard/TopProjectsAndTopTreeSpeciesController.php

```diff
@@ -42,23 +44,15 @@ public function getTopProjects($projects)
 
     public function getTopTreeSpecies($projects)
     {
-        $topSpecies = [];
-
-        $projects->each(function ($project) use (&$topSpecies) {
-            $project->treeSpecies->each(function ($species) use (&$topSpecies) {
-                $topSpecies[] = [
-                    'name' => $species->name,
-                    'amount' => $species->amount,
-                ];
-            });
-        });
-        $speciesCollection = new Collection($topSpecies);
-        $speciesGrouped = $speciesCollection->groupBy('name');
-
-        $sumSpeciesValues = $speciesGrouped->map(function ($species, $name) {
-            return ['name' => $name, 'amount' => $species->sum('amount')];
-        });
-
-        return $sumSpeciesValues->sortByDesc('amount')->take(20)->values()->all();
+        $speciesCollection = TreeSpecies::where('speciesable_type', Project::class)
+            ->whereIn('speciesable_id', $projects->pluck('id'))
+            ->groupBy(DB::raw('BINARY name'))
+            ->groupBy('name')
```

Comment:
I'll approve, but for my own learning, what does this double grouping do? Why is the `BINARY name` needed?

Reply:
This is mainly because name grouping is case sensitive.
In fact, I would summarize this as ->groupBy(DB::raw('BINARY name'), 'name')